modules/xmpp/RtxModifier.js

```diff
@@ -0,0 +1,242 @@
+            } else {
+                logger.info("No previously associated rtx ssrc for " +
+                    " video ssrc " + ssrc);
+                // If there's one in the sdp already for it, we'll just set
+                //  that as the corresponding one
+                let previousAssociatedRtxStream = 
+                    getAssociatedRtxStream (videoMLine, ssrc);
+                if (previousAssociatedRtxStream) {
+                    logger.info("Rtx stream " + previousAssociatedRtxStream + 
+                        " already existed in the sdp as an rtx stream for " +
+                        ssrc);
+                    correspondingRtxSsrc = previousAssociatedRtxStream;
+                } else {
+                    correspondingRtxSsrc = SDPUtil.generateSsrc();
+                    logger.info("Generated rtx ssrc " + correspondingRtxSsrc + 
+                        " for ssrc " + ssrc);
+                }
+                logger.info("Caching rtx ssrc " + correspondingRtxSsrc + 
+                    " for video ssrc " + ssrc);
+                this.correspondingRtxSsrcs[ssrc] = correspondingRtxSsrc;
+            }
+            updateAssociatedRtxStream(
+                videoMLine, 
+                {
+                    id: ssrc,
+                    cname: cname,
+                    msid: msid
+                },
+                correspondingRtxSsrc);
+        });
+        return transform.write(parsedSdp);
+    }
+
+    /**
+     * Remove all reference to any rtx ssrcs that 
+     *  don't correspond to the primary stream.
+     * Must be called *after* extra simulcast streams
+     *  have been removed.
+     */
+    implodeRemoteRtxSsrcs (sdpStr) {
+        let parsedSdp = transform.parse(sdpStr);
+        let videoMLine = 
+            parsedSdp.media.find(mLine => mLine.type === "video");
+        if (videoMLine.direction === "inactive" ||
+                videoMLine.direction === "recvonly") {
+            logger.info("RtxModifier doing nothing, video " +
+                "m line is inactive or recvonly");
+            return sdpStr;
+        }
+        if (!videoMLine.ssrcGroups) {
+            // Nothing to do
+            return sdpStr;
+        }
+
+        // Returns true if the given ssrc is present
+        //  in the mLine's ssrc list
+        let ssrcExists = (ssrcToFind) => {
+            return videoMLine.ssrcs.find((ssrc) => ssrc.id + "" === ssrcToFind);
+        };
+        let ssrcsToRemove = [];
+        videoMLine.ssrcGroups.forEach(group => {
+            if (group.semantics === "FID") {
+                let primarySsrc = group.ssrcs.split(" ")[0];
+                let rtxSsrc = group.ssrcs.split(" ")[1];
+                if (!ssrcExists(primarySsrc)) {
+                    ssrcsToRemove.push(rtxSsrc);
+                }
+            }
+        });
+        videoMLine.ssrcs = videoMLine.ssrcs
+            .filter(ssrc => ssrcsToRemove.indexOf(ssrc.id + "") === -1);
```

Comment:
yeah this does get annoying.  for SdpConsistency i changed everything that returned them or took them in as args to use numbers and always convert away from string.  here there's a bit of a mix because always forcing them felt awkward.  i think part of it comes from the fact that the 'id' field in the sdp ssrc line uses a number, but the ssrc group field has the line as a string of space-separated ssrc strings